.ci/pipelines/e2e-tests-custom.Jenkinsfile

```diff
@@ -0,0 +1,103 @@
+// This library overrides the default checkout behavior to enable sleep+retries if there are errors
+// Added to help overcome some recurring github connection issues
+@Library('apm@current') _
+
+def failedTests = []
+def lib
+
+pipeline {
+
+    agent {
+        label 'linux'
+    }
+
+    options {
+        timeout(time: 600, unit: 'MINUTES')
+    }
+
+    parameters {
+        string(name: "OPERATOR_IMAGE", defaultValue: "", description: "ECK Docker image")
+        string(name: "ELASTICSEARCH_IMAGE", defaultValue: "", description: "Elasticsearch image under test ")
+        string(name: "ELASTICSEARCH_VERSION", defaultValue: "", description: "Elasticsearch version under test")
+        string(name: "KIBANA_IMAGE", defaultValue: "", description: "Kibana image under test ")
+        string(name: "KIBANA_VERSION", defaultValue: "", description: "Kibana version under test")
+        string(name: "ELASTIC_AGENT_IMAGE", defaultValue: "", description: "Elastic Agent image under test ")
+        string(name: "ELASTIC_AGENT_VERSION", defaultValue: "", description: "Elastic Agent version under test")
```

Comment:
Can we add Fleet Server version specifically? Defaults to Elastic Agent version if not specified.